### PR TITLE
Improve: subscript and superscript should now work for all languages and characters.

### PR DESCRIPTION
**Pull request by AtlasAutocode**
## Description

Improve: subscript and superscript should now work for all languages and characters. 

Previous implementation required the use of special open type fonts - normal true type fonts did not show sub or superscript characters. The example program used the Apple SF-UI font that can only be used for apple devices and is not permitted on Windows, Android, Linux or web. The Apple font only implemented these attributes for numbers and lower case english letters.

## Related Issues

Issue 1290 discussed this issue and gave original solution for Apple devices using 'SF-Pro-Display-Regular.otf'. It was extended for use on Android which is not permitted for Apple-only fonts according to the Apple fonts web page.

## Additional notes
Previous implementation required the Apple 'SF-Pro-Display-Regular.otf' font which is only licensed/permitted for use on Apple devices.

I would recommend removing this font from the sample app to prevent users inadvertently using this font on non-Apple devices.

## Checklist

- [X] I read the [Contributor Guide](../CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [X] I titled the PR using [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0).
- [X] I did not modify the `CHANGELOG.md` nor the package version in `pubspec.yaml` files.
- [X] All existing and new tests are passing.
- [X] I have run the commands in `./scripts/before_push.sh` and it all passed successfully

## Breaking Change

Does your PR require developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate that with a `!` in the title as explained in [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0)).
- [X] No, this is *not* a breaking change.

### CODE REVIEW

**Review by singerdmx — APPROVED** [review]

**Comment by singerdmx** [comment]
@ellet0 please take a look when you get a chance

**Comment by ellet0** [comment]
Thank you for your contributions. Can you share a minimal example to reproduce the current issue?

Regarding the apple font in the example, I will update it soon.

**Comment by ellet0** [comment]
> But I put it in in case someone is relying on special fonts that have capabilities that I am not aware of. What do you think?

I'm not aware of all the use cases for the old behavior too, I would usually make it optional and stick to the current behavior and change this in the next major release by document this somewhere or in todos

Although it's completely okay to do it this way and it is usually more preferable, some developers might prefer not to change the behavior much for the working app.

I prefer simplicity over stability and additional complexity when it's not required.

Thank you for contributing once again.

File: lib/src/l10n/generated/quill_localizations_en.dart [hunk 1]
```diff
@@ -128,7 +128,7 @@ class FlutterQuillLocalizationsEn extends FlutterQuillLocalizations {
   String get alignRight => 'Align right';
 
   @override
-  String get justifyWinWidth => 'Justify win width';
```
[the remark this answers is not in the document]
Comment (AtlasAutocode) [reply lib/src/l10n/generated/quill_localizations_en.dart:131]: Yes. I had changed this in a previous PR but did not do it correctly. It was helpful that you pointed out the regenerate_translations script.
On a side note: I am using Android Studio. When I run a script, it runs in a popup command window and the window closes before I can read the results! Do you know any way to redirect the script output to something permanent?

[the remark this answers is not in the document]
Comment (ellet0) [reply lib/src/l10n/generated/quill_localizations_en.dart:131]: > Yes. I had changed this in a previous PR but did not do it correctly. It was helpful that you pointed out the regenerate_translations script.
> On a side note: I am using Android Studio. When I run a script, it runs in a popup command window and the window closes before I can read the results! Do you know any way to redirect the script output to something permanent?

I haven't used Android Studio for a while, although I still use Intellij IDEA, if you go to the run tab at the botton you should be able to find the ourput, the configurations/settings might have been changed to have this behavior, you could run the script in terminal, if you're on Windows you will need Git Bash as this is sh script, the script is very simple, alternatively, you could use the mentioned commands in Translation page in `doc` folder

**Comment by AtlasAutocode** [comment]
This PR should also fix issue 1846

**Comment by singerdmx** [comment]
@AtlasAutocode are you planning to make more changes for this PR or you feel it is in good shape already?

**Comment by ellet0** [comment]
Can we revert `Justify min width` change? 
The key doesn't match the value. Besides, we added new keys instead of changing this in case we want to revert or a new widget that uses it.

**Comment by AtlasAutocode** [comment]
No more changes to this PR planned - ready to go.

justifyWinWidth should give tooltip 'Justify' - look in word processors (such as Word) to see what they use. Please refer to Issue 1806. If you want key and tooltip to match, then change the key to 'justify'. I think the original naming was to confer: 'Justify over the whole window width' but the current abbreviation is too confusing.

**Comment by ellet0** [comment]
> justifyWinWidth should give tooltip 'Justify' - look in word processors (such as Word) to see what they use. Please refer to Issue 1806. If you want key and tooltip to match, then change the key to 'justify'. I think the original naming was to confer: 'Justify over the whole window width' but the current abbreviation is too confusing.

I see however it's not directly related and doesn't change the key for other translations and require changing it, I would recommend creating a key and reverting this change for now.

This way it will be missing in [untranslated.json](https://github.com/singerdmx/flutter-quill/blob/master/lib/src/l10n/untranslated.json) so contributors can know the missing and incomplete.

The current way will give the incorrect text for the other 39 translations and won't update this file.

**Comment by AtlasAutocode** [comment]
OK. Do you want to do it or should I?

Is this the correct syntax for quill_en.arb?

> "justify": "Justify",
   "@justify": {
    "description": "Justify the text over the full window width"
  }

And update  toggle_style_button.dart line 82
>           'justify' => (
>              context.loc.justify,
>              Icons.format_align_justify
>           ),

Perhaps the key should be alignJustify since it is part of the group with alignLeft etc? Tooltip 'Align justify' - would that be clearer?

**Comment by ellet0** [comment]
> OK. Do you want to do it or should I?

It's up to you, I can do it myself if you want.

**Comment by AtlasAutocode** [comment]
Done and submitted.
There are no other usages of this translation key.
There seems to be a conflict in the 'ko' locale which I don't understand - can you review it?

**Comment by ellet0** [comment]
> Done and submitted. There are no other usages of this translation key. There seems to be a conflict in the 'ko' locale which I don't understand - can you review it?

I will take a look at it soon, at the moment I'm moving `rich_clipboard` to be part of `flutter_quill_extensions`

Do you know what this function does?
```dart
Future<void> _update(void Function() listener) async {
    final reader = await SystemClipboard.instance?.read();
    if (reader != null) {
      final available = reader.platformFormats;
      if (_canPaste != available.isNotEmpty) {
        _canPaste = available.isNotEmpty;
        listener();
      }
    }
  }
```
In the file (https://github.com/singerdmx/flutter-quill/blob/master/lib/src/widgets/toolbar/buttons/clipboard_button.dart)

I'm outdated as I haven't looked into the source code for a while and I'm unfamiliar with `ClipboardMonitor`In
- #1843

**Comment by AtlasAutocode** [comment]
_update is called periodically to check the SystemClipboard to see if there is anything that could be pasted into the editor.
It is needed in case, at some future time, something is put on the clipboard from a different program. In this case, the quill editor would have no knowledge that something can now be pasted.
Essentially, it enables visual feedback for paste is now possible. Copy-paste between programs is very common on desktops.

I considered just having the paste button always active. In which case, if nothing can be pasted - give no response (ugh - user wonders why it didn't work) or error message (ugh - translations needed and user must dismiss the message). Visual feedback is simple, intuitive and needs no explanation.

**Comment by ellet0** [comment]
The problem is, that I'm trying to abstract usages of `super_clipboard` and provide some default or disable some features that are not supported:

```dart

/// An abstraction to make it easy to provide different implementations
@immutable
abstract class ClipboardService {
  Future<bool> canProvideHtmlText();
  Future<String?> getHtmlText();

  Future<bool> canProvidePlainText();
  Future<String?> getPlainText();

  Future<bool> canProvideImageFile();
  Future<Uint8List> getImageFileAsBytes();

  Future<bool> canProvideGifFile();
  Future<Uint8List> getGifFileAsBytes();
}

```

The default implementation should only depend on Flutter without any external plugins, throwing an exception or simply ignoring the call will cause unexpected behavior, do you have any suggestions on this?

**Comment by ellet0** [comment]
> There seems to be a conflict in the 'ko' locale which I don't understand - can you review it?

This seems to be a conflict with:
- #1911

As it has been merged and your branch is outdated with the master branch, as you edited the translations and added `alignJustify` key, while Git was able to detect the changes in the source of the translations, it wasn't for the generated dart code, the above PR update `justifyWinWidth` to `좌우로 정렬` in Korean

There are multiple ways to fix this, the simplest is to choose the changes of the #1911 of `justifyWinWidth` above this, while for the `b/src/l10n/untranslated.json` it needs to be regenerated again after fetching the changes of the latest `master` branch from the `upstream` remote

**Comment by AtlasAutocode** [comment]
Conflicts resolved

**Comment by singerdmx** [comment]
@ellet0 could you please take another look? Let's merge it if possible

File: lib/src/l10n/quill_en_US.arb [hunk 2]
```diff
@@ -41,7 +41,7 @@
   "alignLeft": "Align left",
   "alignCenter": "Align center",
   "alignRight": "Align right",
-  "justifyWinWidth": "Justify win width",
+  "justifyWinWidth": "Justify",
```
Comment (ellet0) [inline lib/src/l10n/quill_en_US.arb:44]: Is the change for English US still not reverted? It doesn't match the template arb file and the other translation.

**Comment by AtlasAutocode** [comment]
Reverted

**Comment by ellet0** [comment]
> Reverted

Thanks.

**Review by ellet0 — APPROVED** [review]

**Pull request merged**